File: issuehub-webapp/src/main/java/net/jitix/issuehub/service/IssueTypeService.java
package net.jitix.issuehub.service;

import java.util.List;
import net.jitix.issuehub.entity.IssueType;
import net.jitix.issuehub.exception.AppException;

public interface IssueTypeService {

    public IssueType getIssueType(String issueTypeId) throws AppException;
    
    public List<IssueType> getIssueTypes() throws AppException;
    
}

File: issuehub-webapp/src/main/webapp/static/js/service/IssueTypeService.js
issuehubApp.service("IssueTypeService",
        ['$rootScope', '$location', 'HttpService', 'MessageService',
            function($rootScope, $location, HttpService, MessageService) {

                var self = this;

                this.issueTypes = [];

                this.setIssueTypes = function(issueTypes) {
                    alert(angular.toJson(issueTypes,true));
                    this.issueTypes = angular.fromJson(angular.toJson(issueTypes));
                };

                this.updateIssueTypes = function(successCallback) {
                    //get session from backend
                    HttpService.call('api/issuetypes/', 'GET', {},
                            {
                                successCallback: function(responseData) {
                                    if (responseData) {
                                        self.setIssueTypes(responseData);
                                    }
                                    if (successCallback) {
                                        successCallback(responseData);
                                    }
                                },
                                failureCallback: function(responseData) {

                                }
                            });
                };

                this.getIssueTypes = function() {
                    return this.issueTypes;
                };

                this.getIssueType = function(issueTypeId) {
                    for (var issueType in this.issueTypes) {
                        
                    }
                };

                this.updateIssueTypes();

            }])
